### PR TITLE
Remove old release script and workflows

### PR DESCRIPTION
**Pull request by pbench**
To merge after running the new workflow for a few days/weeks

### CODE REVIEW

**Review by woshilapin — APPROVED** [review]
:point_down: :microphone:

**Review by azime — APPROVED** [review]

**Review by xlqian — APPROVED** [review]

**Review by pbougue — APPROVED** [review]

File: readme_release.md [hunk 1]
```diff
@@ -1,90 +0,0 @@
...
-
-## "Regular" release
-
-This lets the script decide if it's major or minor release.
-The decision is based on the data_version located in source/type/data.cpp
-See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how data_version is managed.
-
-First have a look on github's repo at PRs about to be released https://github.com/hove-io/navitia/pulls?q=is%3Apr+is%3Aclosed+sort%3Aupdated-desc
-* Check that `not_in_changelog` and `hotfix` labels are correct and none is missing on PRs that are gonna be released
-* Check that titles are correct (clear, with the component impacted)
```
Comment (pbougue) [inline readme_release.md]: On second thought, I think we should keep it (almost) and just mention in the end that it's done on Github, not by script anymore.
I'm OK to do a PR on this if it's OK with you.

If not, it should be removed from https://github.com/hove-io/navitia/blob/ea3dea7fb49b7dc7f55b7256be1778f8ff9b04d0/CONTRIBUTING.md?plain=1#L71 too.

Comment (pbougue) [reply readme_release.md]: Done in https://github.com/hove-io/navitia/pull/3907/commits/f71f5d5d67327016eff2cfe77a30fda54aaf4b9d
IMO, we can now merge.

Not planning any additional work on the subject.

**Comment by pbougue** [comment]
I rebased the PR (conflicting with https://github.com/hove-io/navitia/pull/3931 as far as I saw).
A second check on the rebase is welcome @pbench :wink:

**Comment by pbench** [comment]
looks good to me :+1:

**Pull request merged**